token/program/src/processor.rs

```diff
@@ -1709,49 +1711,6 @@ mod tests {
         let account = Account::unpack_unchecked(&account_account.data).unwrap();
         assert_eq!(account.amount, 1000);
 
-        // bogus transfer to self using system accounts.
```

Comment:
Can we replace this with a test that shows that:
1. Bogus transfers to self are now disallowed
2. Valid transfers to self are allowed

Reply:
Yeah sure!  I'll toss a loop around the normal tests so they're rerun as self-transfer

Reply:
https://github.com/solana-labs/solana-program-library/pull/1136/commits/8c42335ea863e79a6dd42aac2b3b705595b3faec